Feat/create company from contact

devklynch opened the pull request:
## Type of Change
- [ x] feature ⛲
- [x ] documentation update 📃

## Description
From the add contacts page, you have the option to add a new company. There is an add new company button that pops up the New Company component. To 

## Motivation and Context
This way if the user wants to add a new contact from a new company they can do it on one page and not redirect to add company, then add contact.
Closes https://github.com/turingschool/tracker-crm/issues/53


## Screenshots (if appr
![create_company](https://github.com/user-attachments/assets/47583e2d-2750-4843-b879-24ac4806bdd1)
opriate):

## Added Test?
- [ x] Yes 🫡 Cypress
- [x ] All previous tests still pass 🥳

## Checklist:
- [x ] My code follows the code style of this project.
- [ x] My change requires a change to the documentation.
- [ x] I have updated the documentation accordingly.

wally-yawn reviewed (changes requested):
Overall this looks great, a few suggestions though:

1. The Add New Company button could use some padding above it so it's more separated from the dropdown.
2. It would be great if on clicking Save in the modal, the modal closes and the dropdown populates the new company.
3. Some additional styling on the X of the modal, might need Erin's opinion on what that should look like

epintozzi commented:
> 1. The Add New Company button could use some padding above it so it's more separated from the dropdown.
Yes I agree. A little bit of padding to add some separation

> 2. It would be great if on clicking Save in the modal, the modal closes and the dropdown populates the new company.
It does appear to add it to the dropdown for me, but I agree, as a user, I'd expect the modal to close upon successful save rather than remain open.

> 3. Some additional styling on the X of the modal, might need Erin's opinion on what that should look like
I'm fine with the X as is.

devklynch commented:
@wally-yawn Made requested updates:

- Added padding to the button
- Upon successful save, the modal will close and then the company name is populated
-Also added that if you click outside of the modal it'll close

gif below to show as well
![Updated_Flow](https://github.com/user-attachments/assets/77e3c56a-c0a7-4b9f-aab5-505e6f2ca6f6)

wally-yawn commented:
Nice job with the updates, looks good now and approving!

sethverrill commented on cypress/e2e/NewContact.cy.js:
```diff
@@ -153,6 +172,84 @@ describe("New Contacts page after logging in", () => {
+      cy.url().should("include", "/contacts");
+      cy.contains("Emma Boots").should("exist");
+      cy.contains("Company C").should("exist");
+    });
```
Really good check here

sethverrill commented on cypress/fixtures/mockContactsData.js:
```diff
@@ -1,75 +1,98 @@
-  ]   
-}
+      id: "3", // Add Alice Green here
+      type: "contacts",
```
Can we remove the Alice Green comment?

sethverrill commented on src/components/contacts/CompanyModal.tsx:
```diff
@@ -0,0 +1,45 @@
+    if (onCompanyCreated) {
+      onCompanyCreated(newCompanyId, newCompanyName);
+    }
+    setIsOpen(false);
```
This whole modal is simple and clean

devklynch replied:
Just updated

sethverrill reviewed (changes requested):
This all looks really good! Put a request to change a comment left in the mock data.
